Marlin/src/module/temperature.cpp

```diff
@@ -1038,6 +1048,10 @@ void Temperature::updateTemperaturesFromRawValues() {
  */
 void Temperature::init() {
 
+  // Flag that the thermalManager should be running
+  if (inited) return;
+  inited = true;
```

Comment:
All this added stuff for resetting the watchdog should only be added for the platforms where it matters. Recommend adding to `Conditionals_post.h` something like this:
```cpp
#define EARLY_WATCHDOG (ENABLED(USE_WATCHDOG) && !defined(__AVR__))
```
…and then wrap these additions in `#if EARLY_WATCHDOG`/`#endif`.

Reply:
I could do it. I do get your point: "Avoid wasting an extra byte of SRAM on platforms that do not need it"...
Dont believe i am happy with the current solution: It is more like a hack, but HW is forcing us to use it...
I believe that on AVR watchdog can be reprogrammed as needed, so this hack is not needed there. On LPC i am not sure...

Reply:
Somehow i also think disabling the watchdog at the setup phase is also some kind of hack: In my specific case, i am using an Arduino DUE on a modified Ramps1.4
SAM3X8E starts with all IO ports configured as inputs with pull ups. So, it is very important to reconfigure the ports as fast as possible, because otherwise, the pullups turn on the Hotend and bed heaters. Watchdog is an important safety measure at all times, not only at loop() time ;)

Reply:
> Avoid wasting an extra byte of SRAM on platforms that do not need it

Well, it's also a minor matter of the general style that Marlin is coded in. The use of compiler directives to leave out unnecessary code not only keeps the binary slimmer, it also acts as a handy reference to what is really required. We can look at the code and see: ah, this is needed, but that is not needed. The reason why the `#include` lines at the top of the G-code handler files are wrapped in `#if` blocks is not just to make the compilation quicker, it also gives us a nice statement at the top of these files about what the real dependencies are. And this gives us some insight into how we might re-organize things as we refactor in the future.

> Watchdog is an important safety measure at all times…

Certainly. And if we discover that terrible things can happen in the earliest stages of boot-up on certain MCUs, and we can use the watchdog to prevent those disasters, then we should. But I haven't seen any situations where –for example– all the heaters get stuck on during boot-up, and I'd be very surprised if that starts to occur in spite of Marlin's thorough pin initializations during setup. I'd be doubly-surprised if the watchdog turns out to be the solution to that kind of issue.

Reply:
I agree. Watchdog is never a solution. It is a safeguard against bugs in sw or hw where is it preferable to reset a regain control rather than losing control of the process